core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -103,16 +108,27 @@ public AbstractClient(Subject subject, InetSocketAddress address,
     mRetryPolicySupplier = retryPolicySupplier;
     mServiceVersion = Constants.UNKNOWN_SERVICE_VERSION;
     // TODO(ggezer) review grpc channel initialization
-    mChannel = GrpcChannelBuilder
-        .forAddress("localhost", 50051)
-        .usePlaintext(true)
-        .build();
+    mChannel = GrpcChannelBuilder.forAddress("localhost", 50051).usePlaintext(true).build();
+    mJobChannel = GrpcChannelBuilder.forAddress("localhost", 50052).usePlaintext(true).build();
+    mVersionService = AlluxioVersionServiceGrpc.newBlockingStub(mChannel);
   }
 
   /**
-   * @return a Thrift service client
+   * @return the type of remote service
    */
-  protected abstract AlluxioService.Client getClient();
+  protected abstract AlluxioServiceType getRemoteServiceType();
+
+  protected long getRemoteServiceVersion() {
+    try {
+      return mVersionService
+              .getServiceVersion(
+                      GetServiceVersionPRequest.newBuilder().setServiceType(getRemoteServiceType()).build())
+              .getVersion();
+    } catch (Exception e) {
+      LOG.error("Failed to get remote service version for type: %s", getRemoteServiceType().name(), e);
```

Comment:
Add string to ExceptionMessage?

Reply:
Being addressed as part of boot-up and security work.